clients/client-python/gravitino/filesystem/gvfs.py

```diff
@@ -362,11 +395,19 @@ def open(
         :param kwargs: Extra args
         :return A file-like object from the filesystem
         """
-        context: FilesetContext = self._get_fileset_context(path)
-        return context.get_fs().open(
-            self._strip_storage_protocol(
-                context.get_storage_type(), context.get_actual_path()
-            ),
+        if mode.startswith("w"):
+            data_operation = FilesetDataOperation.CREATE
+        elif mode.startswith("a"):
+            data_operation = FilesetDataOperation.APPEND
```

Comment:
I'm a little confused why `write` is mapped to `CREATE`. `read` is mapped to `OPEN`.

Reply:
`Write` will open the file, too. `Write` will truncate only if the file exists.

Reply:
The `w` mode here will truncate the existing file before writing, which is equivalent to creating a file before writing. Since we currently only report one data operation when reporting the audit, this corresponds to the `CREATE` operation; The `r` mode is defined as a `read` operation in Python itself.
![image](https://github.com/user-attachments/assets/98816104-e9eb-45a2-b785-8a613598bad1)

Reply:
We cannot easily determine whether the storage system needs to be truncated first, unless we explicitly call the file interface to check, which will bring an additional IO operation. The difference here compared to the Hadoop FileSystem open is that multiple data operations are represented by the same interface, and the actual data operation needs to be determined according to the mode.